src/truncate.jl

```diff
@@ -1,24 +1,71 @@
 """
-    truncated(d::UnivariateDistribution, l::Real, u::Real)
+    truncated(d0::UnivariateDistribution; [lower::Real], [upper::Real])
+    truncated(d0::UnivariateDistribution, lower::Real, upper::Real)
```

Comment:
Should we remove this from the docstring to encourage the use of the kwargs version? Or maybe even remove it and use a `_truncated(d, lower, upper)` for dispatch?

Reply:
> Should we remove this from the docstring to encourage the use of the kwargs version?

I would prefer it if a user who used to write `truncated(Normal(), -2, 2)` didn't now have to use the more verbose `truncated(Normal(); lower=-2, upper=2)`. Not only is it more verbose, but the former was easy to understand, and now every usage of `truncated` before now needs to be updated. We can't really avoid that for the left- and right-truncated versions if we want to discontinue using `Inf` to mean "unbounded".

I don't see using the args version as being a problem; it's using `Inf` to mean "unbounded" that's a problem. The docstring includes the `nothing` signatures, so they are free to just replace all `Inf`s with `nothing`, but I suspect they will eventually adopt the more explicit notation in the left- or right-truncated cases, and hence that's what we recommend in the examples.

Currently the deprecation warning is very strong (which may not be ideal), and I expect users using partial bounds will instantly switch to the new syntax to avoid the annoying warnings. I suppose we could add a note to the documentation about deprecated behavior.

> Or maybe even remove it and use a `_truncated(d, lower, upper)` for dispatch?

I thought about that, but then those wanting to specialize `truncated` for their distribution need to overload something that looks like a private function. Plus there are the points given above.

Reply:
Sounds reasonable. I'm also not a fan of making unexported `_...` functions part of the interface (as unfortunately still the case with `_rand!`...).